### PR TITLE
lookup in http.listen

### PR DESCRIPTION
**Pull request by sirenkovladd**
### What does this PR do?

Fixes: https://github.com/oven-sh/bun/issues/4358
This adds dns.lookup when calling http.Server.listen

(http.Server instanceof net.Server)
https://github.com/nodejs/node/blob/v20.5.1/lib/net.js#L2062-L2073

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

- [X] I ran `make js` and committed the transpiled changes
- [X] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [X] I included a test for the new code, or an existing test covers it

### CODE REVIEW

**Review by paperclover — APPROVED** [review]
looks good. i wish this dns lookup wasnt needed for every node:http server, but it seems this is the behavior node.js does

File: test/js/node/http/node-http.test.ts [hunk 1]
```diff
@@ -823,4 +826,41 @@ describe("node:http", () => {
...
+      if (err) {
+        return closeAndFail("dns lookup failed");
+      }
+
+      server.listen(
+        0,
+        host,
+        mustCall(() => {
+          const address = server.address() as AddressInfo;
+          expect(address.address).toStrictEqual(ip);
```
Comment (Jarred-Sumner) [inline test/js/node/http/node-http.test.ts:920]: this test is failing

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
lets merge once test failures are fixed

**Comment by sirenkovladd** [comment]
@Jarred-Sumner @paperdave 
This binary contains no changes to http.ts [bun-linux-x64](https://github.com/oven-sh/bun/actions/runs/5986240007#:~:text=Artifacts,-Produced%20during%20runtime)
I have change in `src/js/out/InternalModuleRegistryConstants.h` but this is the out file
I really would not like to push changes from this file, as it will be impossible to review

**Comment by paperclover** [comment]
the internal module registry constants is a generated file based on bundling. ci currently isnt able to generate this file, so we commit them locally. that's probably the reason for the test failure

**Comment by sirenkovladd** [comment]
Ok, should I push this file? Are you planning to fix this?

**Comment by paperclover** [comment]
yes, can you push that generated file?

**Comment by sirenkovladd** [comment]
@Jarred-Sumner can you retest?

File: test/js/node/http/node-http.test.ts (hunk 1 above)
Comment (sirenkovladd) [reply test/js/node/http/node-http.test.ts:920]: https://github.com/oven-sh/bun/actions/runs/6002838901/job/16325640437#step:5:6184

**Comment by sirenkovladd** [comment]
@Jarred-Sumner

**Comment by sirenkovladd** [comment]
@Jarred-Sumner @paperdave

**Pull request closed without merging**